middleware/protect.js

```diff
@@ -31,15 +31,15 @@ function forceLogin (keycloak, request, response) {
   }
 
   let uuid = UUID();
-  let loginURL = keycloak.loginUrl(uuid, redirectUrl);
+  let loginURL = (routeName === "register") ? keycloak.registerUrl(redirectUrl) : keycloak.loginUrl(uuid, redirectUrl);
   response.redirect(loginURL);
 }
 
 function simpleGuard (role, token) {
   return token.hasRole(role);
 }
 
-module.exports = function (keycloak, spec) {
+module.exports = function (keycloak, spec, routeName) {
   let guard;
 
   if (typeof spec === 'function') {
```

Comment:
what is this Line doing?

Reply:
This method is being called from index.js protect method and takes keycloak object, spec i.e. some priviledges, roles etc and new parameter routeName i.e login/register. From that parameter we need to decide either redirection requires for login or register page.